jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/content/ContentSinkSubscriber.java

```diff
@@ -76,6 +87,8 @@ public void onError(Throwable failure)
     @Override
     public void onComplete()
     {
-        sink.write(true, null, callback);
+        // wait until called twice: once from last write success and once from the publisher
+        if (lastAndComplete.decrementAndGet() == 0)
+            callback.succeeded();
```

Comment:
This method cannot be called twice, as it violates the spec, for example if a class subclasses `ContentSinkSubscriber`.

Can you detail what is the problem that lead to this fix?
Seems a wrong fix if the publisher of chunks respects reactive streams.